server/auth.go

```diff
@@ -37,6 +38,8 @@ type ClientAuthentication interface {
 	GetTLSConnectionState() *tls.ConnectionState
 	// Optionally map a user after auth.
 	RegisterUser(*User)
+	// RemoteAddress expose the connection information of the client
+	RemoteAddress() *net.TCPAddr
```

Comment:
At the risk of being annoying for users that would have to have an helper function to convert to *net.TCPAddr, would it make more sense to simply return the interface `Addr`, just in case later we have other type of connections than `net.TCPAddr` ones?